engine/gamesys/src/gamesys/test/resource/script_atlas.script

```diff
@@ -136,48 +197,69 @@ local function test_setting_valid(self)
     -- test set wrong path
-    assert_error(function() resource.set_atlas("/missing-path", {}) end)
+    assert_error(function() fn("/missing-path", {}) end)
+
     -- test set wrong texture path
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = "wrong-path", animations = {}, geometries = {}}) end)
+
+    -- test having no animations or geometry table
+    assert_error(function() fn(valid_path, {
+        texture = self.valid_texture_path}) end)
+
     -- test set missing fields in animation
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = self.valid_texture_path, animations = {{}}, geometries = {}}) end)
+
     -- test set missing fields in geometry
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = self.valid_texture_path, animations = {}, geometries = {{}}}) end)
+
     -- test specifying animations but no geometry
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = self.valid_texture_path, animations = {make_anim("test", 1, 1, 1, 2)}, geometries = {}}) end)
+
     -- test specifying incorrect animation start/end
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = self.valid_texture_path, animations = {make_anim("test", 1, 1, 0, 0)}, geometries = {make_quad(1,1,1,1)}}) end)
+
     -- test uneven components in data array
-    assert_error(function() resource.set_atlas(self.valid_tileset_path, {
+    assert_error(function() fn(valid_path, {
         texture = self.valid_texture_path, animations = {}, geometries = { {vertices = {1}, uvs = {}, indices = {}} }}) end)
 end
 
+local function test_create_atlas_invalid(self)
+    -- test creating atlas at a resource path that already exists
+    assert_error(function() resource.create_atlas(self.valid_tileset_path) end)
+end
+
 function init(self)
     self.update_counter = 0
     self.valid_tileset_path = "/tile/valid.t.texturesetc"
     self.valid_texture_path = "/tile/mario_tileset.texturec"
 
-    test_simple(self)
-    test_setting_valid(self)
-    test_fail_all(self)
+    -- valid tests
+    test_create_get_atlas(self)
+    test_create_release_atlas(self)
+    test_get_set_atlas(self)
+    test_set_atlas_valid(self)
+
+    -- invalid tests
+    test_get_atlas_invalid(self)
+
+    -- resource.set_atlas and resource.create_atlas should support the same arguments
+    -- so we can reuse the test functions here
+    test_atlas_invalid(self, resource.set_atlas, self.valid_tileset_path)
+    test_atlas_invalid(self, resource.create_atlas, self.valid_tileset_path)
+
+    -- do a specific invalid test for create_atlas
+    test_create_atlas_invalid(self)
```

Comment:
Added tests for creating atlases as well to this test file instead of creating a whole new test. There is a lot of things in common with create_atlas and set_atlas with a bunch of helper functions so figured it would be better to reuse that instead of a new test